### PR TITLE
docs(roadmap): Adds a proposed roadmap document for wasmCloud projects

### PR DESCRIPTION
**Pull request by thomastaylor312**
I also fixed a license issue where we didn't have the date set

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+see something on here that you think should be on here, feel free to [open a
+PR!](https://github.com/wasmCloud/wasmCloud/pulls)
+
+## Now
+
+This category contains things we are currently (or will very soon be) working on. By definition it
+is very narrow so you know exactly what is coming next 
+
+- SQL V2 contract
+- Finishing up [wadm](https://github.com/wasmCloud/wadm)
```
Comment (brooksmtownsend) [inline ROADMAP.md:21]: If we can flesh these out a little bit, saying that we're doing V2 to further align with the Webassembly standard and protect against injection-type attacks that would be good

Re: wadm, could we add a few high level bullet points like improving the reconciliation algorithm, add a proper error state, and more that you've thought of?

**Review by brooksmtownsend — COMMENTED** [review]
Love it, just a little suggestion for the "Now" section. Once this is merged we can create a project at the org level

**Review by LiamRandall — COMMENTED** [review]
LGTM

**Comment by stephenandary** [comment]
I have questions about LinkDef 2.0 because of incorrectness in how I have been thinking about them that was not valid due to lattice requirements. 

I may be thinking in terms of microservices and this may be something I need to stop doing. 

How I was thinking of a Link Def: 

- I planned to group some actors with specific link defs to an explicitly hosted provider and group the actors, interfaces, and providers as a unit for a feature I wanted to enable for my app. 

- Why I wanted to do this was to be able to assign specific infrastructure to some functionality so that I could scale it more discretely, and make sure all traffic on that provider was related to the specific feature for troubleshooting/logging, transparency, and cost. I also felt it made it easier to understand what was required in code to deploy that feature to keep track of dependencies and maybe make it easier to understand. 

- How infrastructure is purchased would be a large factor in determining if this could actually reduce operational cost, or if compute was just being wasted. I am thinking that the current system can use total traffic to scale instead of worrying about more discrete requests based on features enabled. 

-  An example scenario is a load balancer "feature" that may require an HTTP Provider, the LB actor, and the http-interface. A load balancer could potentially be processing millions of requests per second, and maybe those requests have erroneous data.  My ignorance is related to the pros/cons of having a dedicated HTTP provider host for these requests such that one may have dedicated hardware for the feature. Is there merit to scaling based on more discrete types of traffic? Or does it add complexity and not value?

**Comment by stephenandary** [comment]
Also curious more about provisioning infra. Are there plans for a Terraform Provider or Ansible Playbooks? Etc?

**Comment by thomastaylor312** [comment]
@stephenandary Actually that is a good point. I had infra guides in my head but hadn't captured that in the doc. I'll add it in there

As for linkdefs, I think the grouping stuff you are looking for is more suited to [wadm](https://github.com/wasmCloud/wadm), unless I am totally misunderstanding

**Comment by stephenandary** [comment]
@thomastaylor312 let me try to describe better. 

Within a single lattice is it possible to do option A or only option B? 
This is where the cost, transparency, logging, and troubleshooting questions I am wondering about originate. Is this what you understood? 

Option A: 
``` mermaid
graph TD
    A[Actor A] -->|LD Name = featureA| B{Http Provider}
    C[Actor B] --> |LD Name = featureB|D{Http Provider}
```
vs

Option B: 
``` mermaid
graph TD
    A[Actor A] -->|LD Name = default| B{Http Provider}
    C[Actor B] --> |LD Name = default|B{Http Provider}
```

**Comment by thomastaylor312** [comment]
It is possible to do both of them. You can link with the same link name to the same provider (so a single http provider can be routing calls to multiple actors) or you can have ones separated by features. Between that and a wadm manifest, I think you can leverage things like different groupings of hardware. The linking feature is already there (no extra work needed). Linkdefs 2.0 is making it easier to discover what kind of configuration is needed for a given link definition (as well as handling errors better). Wadm being finished is the full unlock for what I think you are describing

**Review by brooksmtownsend — APPROVED** [review]

**Review by mattwilkinsonn — APPROVED** [review]
LGTM! One collaboration question - do we want to keep this in the repository here?  I'm thinking this might be hard to comment on / have multiple people editing in the future.

Perhaps a GitHub Project could work for this? Or a Trello board / Google doc etc. Project is probably best for discoverability purposes.

**Pull request merged**